src/libcrun/container.c

```diff
@@ -839,14 +839,33 @@ static int
 libcrun_configure_handler (struct container_entrypoint_s *args, libcrun_error_t *err)
 {
   const char *annotation;
-
   annotation = find_annotation (args->container, "run.oci.handler");
-  /*Nothing to do.  */
+
+  /* Fail with EACCESS if global handler is already configured and there was a attempt to override it via spec. */
+  if (args->context->handler != NULL && annotation != NULL)
+    {
+      return crun_make_error (err, EACCES, "invalid attempt to override already configured global handler: %s", args->context->handler);
+    }
+
+    /* In selection order global_handler takes more priority over handler configured via spec annotations. */
+    /* Check if crun is being invoked as krun via global_handler. */
+#if HAVE_DLOPEN && HAVE_LIBKRUN
+  if (args->context->handler != NULL && (strcmp (args->context->handler, "krun") == 0))
+    {
+      return libcrun_configure_libkrun (args, err);
+    }
+#endif
+
+  /* Do nothing: no annotations or global_handler configured */
   if (annotation == NULL)
     return 0;
 
   if (strcmp (annotation, "krun") == 0)
-    return libcrun_configure_libkrun (args, err);
+    {
+      /* set global_handler equivalent to "krun" so that we can mount kvm device */
+      args->context->handler = "krun";
+      return libcrun_configure_libkrun (args, err);
+    }
```

Comment:
this code chunk looks very similar to the previous one.  Any chance we could refactor the code to not have the same check twice?